src/biogeophys/SoilTemperatureMod.F90

```diff
@@ -1372,12 +1445,14 @@ subroutine Phasechange_beta (bounds, num_nolakec, filter_nolakec, dhsdT, &
                         end if
                      endif  ! end of heatr > 0 if-block
 
-                     if (j >= 1) then
-                        xmf(c) = xmf(c) + hfus*(wice0(c,j)-h2osoi_ice(c,j))/dtime
+                     if (j >= 1) then !why before it was two same statements?
```

Comment:
I'm not sure what your question is here? Can you elaborate? I want to make sure we address any issues that you find the code confusing.

Reply:
This block looks like this in master:
```
                     if (j >= 1) then
                        xmf(c) = xmf(c) + hfus*(wice0(c,j)-h2osoi_ice(c,j))/dtime
                     else
                        xmf(c) = xmf(c) + hfus*(wice0(c,j)-h2osoi_ice(c,j))/dtime
                     endif
```
I was unsure if the condition was redundant or not.

Reply:
Made an issue to look into this #1818. Since, you do need the if statement we might resolve this just by removing your comment.

Reply:
We removed the comment. The issue doesn't need to be resolved anymore since in excess ice there is a difference. With excess ice there is an additional term added in for the j>=1 case. It would be better to have the equation the same, but then add the extra term for j>=1 case. But, we'll forget about that for now.